airflow/providers/google/cloud/hooks/looker.py

```diff
@@ -0,0 +1,268 @@
+            )
+        if status == JobStatus.CANCELLED.value:
+            raise AirflowException(f'PDT materialization job was cancelled. Job id: {materialization_id}.')
+        if status == JobStatus.UNKNOWN.value:
+            raise AirflowException(
+                f'PDT materialization job has unknown status. Job id: {materialization_id}.'
+            )
+
+        self.log.info('PDT materialization job completed successfully. Job id: %s.', materialization_id)
+
+    def get_api_source(self):
+        """Returns origin of the API call."""
+
+        # Infer API source based on Airflow version, e.g.:
+        # "2.1.4+composer" -> Cloud Composer
+        # "2.1.4" -> Airflow
+
+        self.log.debug(
+            "Got the following version for connection id: '%s'. Version: '%s'.", self.looker_conn_id, version
+        )
+
+        return 'composer' if 'composer' in version else 'airflow'
+
+    def get_looker_sdk(self):
+        """Returns Looker SDK client for Looker API 4.0."""
+
+        conn = self.get_connection(self.looker_conn_id)
+        settings = LookerApiSettings(conn)
+
+        transport = requests_transport.RequestsTransport.configure(settings)
+        return methods40.Looker40SDK(
+            auth_session.AuthSession(settings, transport, serialize.deserialize40, "4.0"),
+            serialize.deserialize40,
+            serialize.serialize,
+            transport,
+            "4.0",
+        )
+
+
+class LookerApiSettings(api_settings.ApiSettings):
+    """Custom implementation of Looker SDK's `ApiSettings` class."""
+
+    def __init__(
+        self,
+        conn: Connection,
+    ) -> None:
+        self.conn = conn  # need to init before `read_config` is called in super
+        super().__init__()
+
+    def read_config(self) -> Dict[str, str]:
+        """
+        Overrides the default logic of getting connection settings. Fetches
+        the connection settings from Airflow's connection object.
+        """
+
+        config = {}
+
+        if self.conn.host is None:
+            raise AirflowException(f'No `host` was supplied in connection: {self.conn.id}.')
+
+        if self.conn.port:
+            config["base_url"] = f"{self.conn.host}:{self.conn.port}"  # port is optional
+        else:
+            config["base_url"] = self.conn.host
+
+        if self.conn.login:
+            config["client_id"] = self.conn.login
+        else:
+            raise AirflowException(f'No `login` was supplied in connection: {self.conn.id}.')
+
+        if self.conn.password:
+            config["client_secret"] = self.conn.password
+        else:
+            raise AirflowException(f'No `password` was supplied in connection: {self.conn.id}.')
+
+        extras = self.conn.extra_dejson  # type: Dict
+
+        if 'verify_ssl' in extras:
+            config["verify_ssl"] = extras["verify_ssl"]  # optional
+
+        if 'timeout' in extras:
+            config["timeout"] = extras["timeout"]  # optional
```

Comment:
The connection is configured very often as a URL, which unfortunately makes the field type to be text. Should we here be casting type? See: https://github.com/apache/airflow/pull/21155

Reply:
Our Looker SDK can take str and will convert it to int. See: https://github.com/looker-open-source/sdk-codegen/blob/2432a688438d64d18531ee3ac920c9933bc8a185/python/looker_sdk/rtl/api_settings.py#L90